packages/data-context/src/actions/MigrationActions.ts

```diff
@@ -1,22 +1,195 @@
+  getIntegrationFolder,
+  isDefaultTestFiles,
+  getComponentTestFilesGlobs,
+  getComponentFolder,
+  getIntegrationTestFilesGlobs,
+  getSpecPattern,
+} from '../sources/migration'
+import EventEmitter from 'events'
+import type { CypressError } from '@packages/errors'
+
+class LegacyPluginsIpc extends EventEmitter {
+  constructor (readonly childProcess: ChildProcess) {
+    super()
+    childProcess.on('message', (msg: { event: string, args: any[] }) => {
+      this.emit(msg.event, ...msg.args)
+    })
+
+    childProcess.once('disconnect', () => {
+      this.emit('disconnect')
+    })
+  }
+
+  send(event: 'loadLegacyPlugins', legacyConfig: LegacyCypressConfigJson): boolean
+  send (event: string, ...args: any[]) {
+    if (this.childProcess.killed) {
+      return false
+    }
+
+    return this.childProcess.send({ event, args })
+  }
+
+  on(event: 'ready', listener: () => void): this
+  on(event: 'loadLegacyPlugins:error', listener: (error: CypressError) => void): this
+  on(event: 'childProcess:unhandledError', listener: (legacyConfig: LegacyCypressConfigJson) => void): this
+  on(event: 'loadLegacyPlugins:reply', listener: (legacyConfig: LegacyCypressConfigJson) => void): this
+  on (evt: string, listener: (...args: any[]) => void) {
+    return super.on(evt, listener)
+  }
+}
+
+export async function processConfigViaLegacyPlugins (projectRoot: string, legacyConfig: LegacyCypressConfigJson): Promise<LegacyCypressConfigJson> {
+  const pluginFile = legacyConfig.pluginsFile ?? await tryGetDefaultLegacyPluginsFile(projectRoot)
+
+  return new Promise((resolve, reject) => {
+    // couldn't find a pluginsFile
+    // just bail with initial config
+    if (!pluginFile) {
+      return resolve(legacyConfig)
+    }
+
+    const cwd = path.join(projectRoot, pluginFile)
+
+    const childOptions: ForkOptions = {
+      stdio: 'inherit',
+      cwd: path.dirname(cwd),
+      env: process.env,
+    }
+
+    const configProcessArgs = ['--projectRoot', projectRoot, '--file', cwd]
+    const CHILD_PROCESS_FILE_PATH = require.resolve('@packages/server/lib/plugins/child/require_async_child')
+    const ipc = new LegacyPluginsIpc(fork(CHILD_PROCESS_FILE_PATH, configProcessArgs, childOptions))
+
+    ipc.on('ready', () => {
+      ipc.send('loadLegacyPlugins', legacyConfig)
+    })
+
+    ipc.on('loadLegacyPlugins:reply', (modifiedLegacyConfig) => {
+      resolve(modifiedLegacyConfig)
+      ipc.childProcess.kill()
+    })
+
+    ipc.on('loadLegacyPlugins:error', (error) => {
+      reject(error)
+      ipc.childProcess.kill()
+    })
+
+    ipc.on('childProcess:unhandledError', (error) => {
+      reject(error)
+    })
+  })
+}
 
 export class MigrationActions {
   constructor (private ctx: DataContext) { }
 
+  async initialize (config: LegacyCypressConfigJson) {
+    const legacyConfigForMigration = await this.setLegacyConfigForMigration(config)
+
+    // for testing mainly, we want to ensure the flags are reset each test
+    this.resetFlags()
```

Comment:
Should we add this to `clearCurrentProject`?

Reply:
Hm - I don't think so. `clearCurrentProject` is involved with getting rid of everything to do with the current project, `resetFlags` is called by `MigrationActions#initialize` which is called by `setCurrentProject`. This seems like a reasonable place to reset the state.